logger.go

```diff
@@ -0,0 +1,175 @@
+}
+
+// Warningln is equivalent to the global Warningln function, with the addition of prefix and data content from this Logger.
+func (l Logger) Warningln(args ...interface{}) {
+	l.println(warningLog, l.extendWithPfx(args)...)
+}
+
+// WarninglnWithDepth is equivalent to the global WarninglnWithDepth function, with the addition of prefix and data content from this Logger.
+func (l Logger) WarninglnWithDepth(extraDepth int, args ...interface{}) {
+	l.printlnWithDepth(warningLog, extraDepth, l.extendWithPfx(args)...)
+}
+
+// Warningf is equivalent to the global Warningf function, with the addition of prefix and data content from this Logger.
+func (l Logger) Warningf(format string, args ...interface{}) {
+	l.printf(warningLog, l.pfx(format), l.extend(args)...)
+}
+
+// WarningfWithDepth is equivalent to the global WarningfWithDepth function, with the addition of prefix and data content from this Logger.
+func (l Logger) WarningfWithDepth(extraDepth int, format string, args ...interface{}) {
+	l.printfWithDepth(warningLog, extraDepth, l.pfx(format), l.extend(args)...)
+}
+
+// Error is equivalent to the global Error function, with the addition of prefix and data content from this Logger.
+func (l Logger) Error(args ...interface{}) {
+	l.print(errorLog, l.extendWithPfx(args)...)
+}
+
+// ErrorWithDepth is equivalent to the global ErrorWithDepth function, with the addition of prefix and data content from this Logger.
+func (l Logger) ErrorWithDepth(extraDepth int, args ...interface{}) {
+	l.printWithDepth(errorLog, extraDepth, l.extendWithPfx(args)...)
+}
+
+// Errorln is equivalent to the global Errorln function, with the addition of prefix and data content from this Logger.
+func (l Logger) Errorln(args ...interface{}) {
+	l.println(errorLog, l.extendWithPfx(args)...)
+}
+
+// ErrorlnWithDepth is equivalent to the global ErrorlnWithDepth function, with the addition of prefix and data content from this Logger.
+func (l Logger) ErrorlnWithDepth(extraDepth int, args ...interface{}) {
+	l.printlnWithDepth(errorLog, extraDepth, l.extendWithPfx(args)...)
+}
+
+// Errorf is equivalent to the global Errorf function, with the addition of prefix and data content from this Logger.
+func (l Logger) Errorf(format string, args ...interface{}) {
+	l.printf(errorLog, l.pfx(format), l.extend(args)...)
+}
+
+// ErrorfWithDepth is equivalent to the global ErrorfWithDepth function, with the addition of prefix and data content from this Logger.
+func (l Logger) ErrorfWithDepth(extraDepth int, format string, args ...interface{}) {
+	l.printfWithDepth(errorLog, extraDepth, l.pfx(format), l.extend(args)...)
+}
+
+func (l Logger) extendWithPfx(args []interface{}) []interface{} {
+	if l.prefix != "" {
+		args = append([]interface{}{
+			l.prefix,
+		}, args...)
+	}
+	return l.extend(args)
+}
+
+func (l Logger) extend(args []interface{}) []interface{} {
+	for _, d := range l.data {
+		args = append(args, data{d})
```

Comment:
oh, I missed that (searched for 'data('). Why not use the Data() here? it makes perfect sense, especially if we will be making changes to Data() down the road.

Reply:
Sounds good, but I still don't want to change the current behavior re: nesting. The more change we make to existing behavior now, the more likely we are to break something.